src/olympia/core/apps.py

```diff
@@ -125,18 +125,26 @@ def static_check(app_configs, **kwargs):
 def db_charset_check(app_configs, **kwargs):
     errors = []
 
-    with connection.cursor() as cursor:
-        cursor.execute("SHOW VARIABLES LIKE 'character_set_database';")
-        result = cursor.fetchone()
-        if result[1] != settings.DB_CHARSET:
-            errors.append(
-                Error(
-                    'Database charset invalid. '
-                    f'Expected {settings.DB_CHARSET}, '
-                    f'recieved {result[1]}',
-                    id='setup.E005',
+    try:
```

Comment:
If you "stop" the mysql service and run

```bash
./manage.py monitors --service olympia_database
```

The command raises here as it is a check. we want to return the error object which means that check should not raise.

In general, this django check should also not raise but return the error object.